sdk/servicebus/Azure.Messaging.ServiceBus/src/Plugins/ServiceBusPipeline.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading.Tasks;
+using Azure.Messaging.ServiceBus.Diagnostics;
+using Azure.Messaging.ServiceBus.Plugins;
+
+namespace Azure.Messaging.ServiceBus
+{
+    internal class ServiceBusPipeline
+    {
+        private ReadOnlyMemory<ServiceBusPlugin> _pipeline;
+
+        public ServiceBusPipeline(ReadOnlyMemory<ServiceBusPlugin> pipeline)
```

Comment:
Do we need this level of efficiency in this context, where taking the readability and usability hit of `ReadOnlyMemory` over a list or array segment is justified?

Reply:
ReadOnlyMemory seems like a usability improvement over ArraySegment. Allocating a new array or list for each plugin in the chain would be pretty wasteful.